custom_components/bmw_connected_drive/sensor.py

```diff
@@ -80,6 +80,35 @@ def convert_and_round(
         unit_type=PERCENTAGE,
         device_class=SensorDeviceClass.BATTERY,
     ),
+    "ac_current_limit": BMWSensorEntityDescription(
+        key="ac_current_limit",
+        name="AC current limit",
+        key_class="charging_profile",
+        unit_type=UnitOfElectricCurrent.AMPERE,
+        icon="mdi:current-ac",
+        entity_registry_enabled_default=False,
+    ),
+    "charging_target": BMWSensorEntityDescription(
+        key="charging_target",
+        name="Charging target",
+        key_class="fuel_and_battery",
+        unit_type=PERCENTAGE,
+        icon="mdi:battery-charging-high",
+    ),
+    "charging_mode": BMWSensorEntityDescription(
+        key="charging_mode",
+        name="Charging mode",
+        key_class="charging_profile",
+        icon="mdi:ev-station",
+        value=lambda x, y: x.value,
+    ),
+    "charging_preferences": BMWSensorEntityDescription(
+        key="charging_preferences",
+        name="Charging preferences",
+        key_class="charging_profile",
+        icon="mdi:ev-station",
+        value=lambda x, y: x.value,
```

Comment:
Yeah I agree that charging preference doesn't add much value. For my car it's `NO_PRESELECTION` when the mode is on `IMMEDIATE_CHARGING` or otherwise it's `CHARGING_WINDOW`.